Test/MergeFile.Tests.ps1

```diff
@@ -0,0 +1,303 @@
+    }
+}
+
+Describe 'MergeFile.when destination file exists and user wants to clear it' {
+    It 'should clear existing contents' {
+        Init
+        GivenFile 'merged.txt' -WithContent 'i should disappear'
+        GivenFile 'one.txt' -WithContent 'snafu'
+        WhenMerging 'one.txt' -Into 'merged.txt' -Clear
+        ThenFile 'one.txt' -Exists
+        ThenFile 'merged.txt' -HasContent 'snafu'
+    }
+}
+
+Describe 'MergeFile.when destination file is outside the build root' {
+    It 'should fail' {
+        Init
+        GivenFile 'one.txt' -WithContent 'failed!'
+        WhenMerging 'one.txt' -Into '..\somefile.txt' -ErrorAction SilentlyContinue
+        ThenFailed -WithError 'which\ is\ outside\ the\ build\ root' 
+    }
+}
+
+Describe 'MergeFile.when deleting originals' {
+    It 'should delete source files' {
+        Init
+        GivenFile 'one.txt' -WithContent 'one'
+        GivenFile 'two.txt' -WithContent 'two'
+        WhenMerging 'one.txt','two.txt' -Into 'merged.txt' -AndDeletingSourceFiles
+        ThenFile 'one.txt' -DoesNotExist
+        ThenFile 'two.txt' -DoesNotExist
+        ThenFile 'merged.txt' -HasContent 'onetwo'
+    }
+}
+
+Describe 'MergeFile.when customizing separator' {
+    It 'should separate file contents with separator' {
+        Init
+        GivenFile 'one.txt' -WithContent 'one'
+        GivenFile 'two.txt' -WithContent 'two'
+        WhenMerging 'one.txt','two.txt' -Into 'merged.txt' -WithSeparator ('$(NewLine)') 
+        ThenFile 'merged.txt' -HasContent ('one{0}two' -f [Environment]::NewLine)
+    }
+}
+
+Describe 'MergeFile.when customizing separator and keeping destination file content' {
+    It 'should add separator after existing contents' {
+        Init
+        GivenFile 'one.txt' -WithContent 'one'
+        GivenFile 'two.txt' -WithContent 'two'
+        GivenFile 'merged.txt' -WithContent 'i was here first'
+        WhenMerging 'one.txt','two.txt' -Into 'merged.txt' -WithSeparator ('$(NewLine)') 
+        ThenFile 'merged.txt' -HasContent ('i was here first{0}one{0}two' -f [Environment]::NewLine)
+    }
+}
+
+Describe 'MergeFile.when customizing separator and clearing destination file content' {
+    It 'should not add separator to beginning of file' {
+        Init
+        GivenFile 'one.txt' -WithContent 'one'
+        GivenFile 'two.txt' -WithContent 'two'
+        GivenFile 'merge.tst' -WithContent 'gone gone gone'
+        WhenMerging 'one.txt','two.txt' -Into 'merged.txt' -WithSeparator ('$(NewLine)') -Clear
+        ThenFile 'merged.txt' -HasContent ('one{0}two' -f [Environment]::NewLine)
+    }
+}
+
+Describe 'MergeFile.when files are empty' {
+    It 'should separate file contents with separator' {
+        Init
+        GivenFile 'one.txt' 
+        GivenFile 'two.txt' 
+        WhenMerging 'one.txt','two.txt' -Into 'merged.txt' 
+        ThenFile 'merged.txt' -HasContent ('')
+    }
+}
+
+Describe 'MergeFile.when files are binary' {

```

Comment:
Maybe also a test for files text file that are not `UTF-8` encoded? E.g. `UTF-8 with BOM` or `UTF-16`

Reply:
This test exists to ensure that the task doesn't use or switch to using a TextReader or the `IO.File.*AllText` . It should operate on files without changing a single byte in them. Adding a test with a UTF-16-encoded file wouldn't do more than this test does.